src/krisi/evaluate/metric.py

```diff
@@ -88,11 +90,50 @@ def evaluate(self, y: Targets, predictions: Predictions) -> None:
             result = e
         self.__safe_set(result, key="result")
 
+    def evaluate_in_group(self, *args) -> "Metric":
+        assert (
+            self.func_group is not None
+        ), "Group Function has to be set to be able to `evaluate_in_group`."
+        try:
+            result = self.func_group(*args, **self.parameters)
+        except Exception as e:
+            result = e
+        self.__safe_set(result, key="result")
+
+        return self
+
+    def evaluate_over_time_in_group(
+        self, *args, window: Optional[int] = None
+    ) -> "Metric":
+        assert (
+            self.func_group is not None
+        ), "Group Function has to be set to be able to `evaluate_in_group`."
+        try:
+            if window is not None:
+                result_rolling = [
+                    self.func_group(
+                        *[el[i : i + window] for el in args],
+                        **self.parameters,
+                    )
+                    for i in range(0, len(args[0]) - 1, window)
+                ]
+            else:
+                # expanding
+                result_rolling = [
+                    self.func_group(*[el[: i + 1] for el in args], **self.parameters)
+                    for i in range(len(args[0]) - 1)
```

Comment:
I don't entirely understand all of this, but I think later, we'll need to support step_size for both rolling window and expanding window, and set a sensible default as well. that's not part of this, right?:)

Reply:
There is no step size now, yes! You can only set window size right now. 
What this does is cut each arg to the correct size, so you can pass in arbitrary amount of args.